<!-- Run the fiducial_detect node -->
<launch>
  <!-- namespace for camera input -->
  <arg name="camera" default="/camera"/>
  <arg name="image" default="image_rect"/>
  <arg name="undistort_points" default="false"/>

  <!-- fiducial_len common values default 0.145, mjWall 0.09  mjGar 0.135 wayne 0.200 -->
  <arg name="fiducial_len" value="$(optenv FIDUCIAL_LEN 0.145)" />

  <node pkg="fiducial_detect" name="fiducial_detect"
    type="fiducial_detect" output="screen" respawn="true">
    <param name="tag_height" value="$(env HOME)/.ros/fiducials/Tag_Heights.xml" />
    <param name="data_directory" value="$(env HOME)/.ros/fiducials" />
    <param name="map_frame" value="odom" />
    <param name="pose_frame" value="slam" />
    <param name="publish_images" value="true" />
    <param name="publish_tf" value="false" />
    <param name="fiducial_len" value="$(arg fiducial_len)"/>
    <param name="undistort_points" value="$(arg undistort_points)"/>
    <remap from="fiducial_detect/camera" to="$(arg camera)/$(arg image)"/>
    <remap from="fiducial_detect/camera_info" to="$(arg camera)/camera_info"/>
  </node>
</launch>
